tools/src/main/java/org/apache/kafka/tools/streams/StreamsGroupCommand.java

```diff
@@ -398,8 +407,201 @@ protected Admin createAdminClient(Map<String, String> configOverrides) throws IO
+                        if (state == GroupState.DEAD) {
+                            failed.put(groupId, new IllegalStateException("Streams group '" + groupId + "' group state is DEAD."));
+                        } else if (state != GroupState.EMPTY) {
+                            failed.put(groupId, new GroupNotEmptyException("Streams group '" + groupId + "' is not EMPTY."));
+                        }
+                    });
+                }
+            }
+            return failed;
+        }
+
+        List<GroupListing> listDetailedStreamsGroups() {
+            try {
+                ListGroupsResult result = adminClient.listGroups(new ListGroupsOptions()
+                    .timeoutMs(opts.options.valueOf(opts.timeoutMsOpt).intValue())
+                    .withTypes(Set.of(GroupType.STREAMS)));
+                Collection<GroupListing> listings = result.all().get();
+                return listings.stream().toList();
+            } catch (InterruptedException | ExecutionException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        private void printInternalTopicErrors(Map<String, Throwable> internalTopicsDeletionFailures,
+                                              Set<String> deletedGroupIds,
+                                              Set<String> groupIdsWithInternalTopics) {
+            if (!deletedGroupIds.isEmpty()) {
+                if (internalTopicsDeletionFailures.isEmpty()) {
+                    List<String> successfulGroups = deletedGroupIds.stream()
+                        .filter(groupIdsWithInternalTopics::contains)
+                        .collect(Collectors.toList());
+                    System.out.println("Deletion of associated internal topics of the streams groups ('" +
+                        String.join("', '", successfulGroups) + "') was successful.");
+                } else {
+                    System.out.println("Deletion of some associated internal topics failed:");
+                    internalTopicsDeletionFailures.forEach((group, error) ->
+                        System.out.println("* Internal topics of the streams group '" + group + "' could not be deleted due to: " + error));
+                }
+            }
+        }
+
+        // Visibility for testing
+        Map<String, List<String>> retrieveInternalTopics(List<String> groupIds) {
+            Map<String, List<String>> groupToInternalTopics = new HashMap<>();
+            try {
+                Map<String, StreamsGroupDescription> descriptionMap = adminClient.describeStreamsGroups(groupIds).all().get();
+                for (StreamsGroupDescription description : descriptionMap.values()) {
+
+                    List<String> nonInternalTopics = description.subtopologies().stream()
+                        .flatMap(subtopology -> Stream.concat(
+                            subtopology.sourceTopics().stream(),
+                            subtopology.repartitionSinkTopics().stream()))
```

Comment:
This is not correct. I think in the previous version you only filtered out sourceTopics, which was correct. repartitionSinkTopics are _always_ internal topics, and must also be repartitionSourceTopics (of some other subtopology).